// EXCEPTION

// Try Catch block
try {
    File testFile
    testFile.append('Line 5')
}
catch (Exception ex) {
    println('I Catch Everything')
} 


//// Try with multiple catch block
//try {
//    File testFile
//    testFile.append('Line 5')
//}
//catch (NullPointerException ex) {
//    // Prints exception
//    println(ex.toString())
//    // Prints error message
//    println(ex.getMessage())
//} 
//catch (Exception ex) {
//    println('I Catch Everything')
//} 


//// Try Catch Finally block
//try {
//    File testFile
//    testFile.append('Line 5')
//}
//catch (Exception ex) {
//    println('I Catch Everything')
//} 
//finally { 
//    println('I perform clean up')
//}


//// Try Finally block
//try {
//    File testFile
//    testFile.append('Line 5')
//}
//finally { 
//    println('I perform clean up')
//}

